security/pkg/nodeagent/util/util.go

```diff
@@ -70,23 +74,29 @@ func OutputKeyCertToDir(dir string, privateKey, certChain, rootCert []byte) erro
 	if len(dir) == 0 {
 		return nil
 	}
+
+	certFileMode := os.FileMode(0600)
+	if k8sInCluster.Get() != "" {
+		// If this is running on k8s, give more permission to the file certs.
```

Comment:
can you update the comment to explain *why* we do this?

Reply:
Done.